src/lib/repository/Upsert.ts

```diff
@@ -4,41 +4,84 @@ import { RecordResult } from 'jsforce';
-        Util.showSpinner('-- Upserting ' + sObjectName);
+        const messageString = '-- Upserting ' + sObjectName;
+        Util.showSpinner(messageString);
         
         const externalIdString = (sObjectName.startsWith('enxB2B__') ? 'enxB2B__TECH_External_Id__c' : 'enxCPQ__TECH_External_Id__c');
         let sobjectsResult:Array<RecordResult> = new Array<RecordResult>();
 
+        let failedResults: Array<RecordResult> = new Array<RecordResult>();
+        let failedRecords: Array<any> = new Array<any>();
+        
+        let warnings: Array<any> = new Array<any>();
+
         if (records.length < 200) {
             // @ts-ignore: Don't know why, but TypeScript doesn't use the correct method override
             sobjectsResult = await connection.sobject(sObjectName).upsert(records, externalIdString, {}, (err, rets:RecordResult[]) => {
                 if (err) { 
                     Util.log(err); 
                 }
             });
+
+            failedResults = sobjectsResult.filter(e => e.success === false);
+            if (failedResults.length > 0) {
+                failedRecords.push(...records);
+            }
+
         } else {
             // @ts-ignore: Don't know why, but TypeScript doesn't use the correct method override
             sobjectsResult = await this.upsertBulkData(connection, records, sObjectName);
+
+            failedResults = sobjectsResult.filter(e => e.success === false);
+            if (failedResults.length > 0) {
+                //@TO-DO : filter only the records that were failed - after the failed IDs are added to reports
+                failedRecords.push(...records);
+            }
+
         }
 
-        const successCount = sobjectsResult.map((r):number => {return (r.success ? 1 : 0)})
-                                            .reduce((prevVal:number, nextVal:number) => {return prevVal+nextVal});
+        const successCount = sobjectsResult.map((r): number => { return (r.success ? 1 : 0) })
+            .reduce((prevVal: number, nextVal: number) => { return prevVal + nextVal });
 
-        const errorCount = sobjectsResult.map((r):number => {return (r.success ? 0 : 1)})
-                                            .reduce((prevVal:number, nextVal:number) => {return prevVal+nextVal});
+        const errorCount = sobjectsResult.map((r): number => { return (r.success ? 0 : 1) })
+            .reduce((prevVal: number, nextVal: number) => { return prevVal + nextVal });
 
-        
-        await Util.hideSpinner(' done. Success: ' + successCount + ', errors: ' + errorCount);
 
+        const errors = sobjectsResult.filter((elem) => {
+            return elem.success === false;
+        }).map((elem) => {
+            return { "error": elem['errors'] };
+        });
+
+        const initialTabbing = (messageString.length > 21) ? (messageString.length > 29) ? (messageString.length > 37) ? (messageString.length > 45) ?  '\t' : '\t\t' : '\t\t\t' : '\t\t\t\t' : '\t\t\t\t\t';
```

Comment:
W zasadzie to takie _pritifajowanie_ logów to lepiej jakby się działo już w utilsach tu trochę śmieci w tym miejscu kodu